### PR TITLE
feat: add app protocol 

### PR DESCRIPTION
**Pull request by OfTheDelmer**
## Why

We'd like to be able specify http2 when building an http tunnel

## How

Add `app_protocol` to the builder and plumb the `forwards_proto` into the session rpc.

## Validation

Run an example with the http2 app protocol and confirm the correct `forwardsProto` was set to `"http2"` or `""` .

### CODE REVIEW

File: ngrok/src/config/labeled.rs [hunk 1]
```diff
@@ -90,6 +99,15 @@ impl LabeledTunnelBuilder {
...
     }
 
+    /// Sets the L7 protocol string for this tunnel.
+    pub fn app_protocol(&mut self, forwards_proto: impl Into<String>) -> &mut Self {
+        let proto_str = forwards_proto.into();
+        if let Ok(proto) = AppProtocol::from_str(&proto_str) {
+            self.options.common_opts.forwards_proto = proto;
+        }
+        self
+    }
```
Comment (jrobsonchase) [inline ngrok/src/config/labeled.rs:109]: Ditto the previous comment about string/enum/Result

**Review by jrobsonchase — COMMENTED** [review]
After looking over this, I'm less sure about the enum approach. We might want to make it a string all the way through and defer validation to our hosted services. That way we can add new protocols we support there and don't have to go and update the sdk for each one. If we took that approach, the only thing I'd do in the builder methods would be to `.to_lowercase()` it.

Ultimately up to product though. @salilsub thoughts?

**Comment by jrobsonchase** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #130** <a href="https://app.graphite.dev/github/pr/ngrok/ngrok-rust/130?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #131** <a href="https://app.graphite.dev/github/pr/ngrok/ngrok-rust/131?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ngrok/ngrok-rust/130?utm_source=stack-comment).

**Comment by salilsub** [comment]
I think as a general rule we should push more validation to our hosted services so we can adjust those rules over time. There will be exceptions, but I think @jrobsonchase is right on this one.

File: ngrok/src/config/http.rs [hunk 2]
```diff
@@ -137,6 +137,14 @@ impl TunnelConfig for HttpOptions {
             .clone()
             .unwrap_or(default_forwards_to().into())
     }
+
+    fn forwards_proto(&self) -> String {
+        self.common_opts
+            .forwards_proto
+            .clone()
+            .unwrap_or(String::new())
```
Comment (jrobsonchase) [inline ngrok/src/config/http.rs:145]: clippy will probably yell about this. If it does, `unwrap_or_default` instead. If not, leave it :stuck_out_tongue:

**Review by jrobsonchase — APPROVED** [review]

**Pull request merged**